bgpd/bgp_routemap.c

```diff
@@ -2095,9 +2095,21 @@ static void *route_set_lcommunity_delete_compile(const char *arg)
 
 	rcom = XCALLOC(MTYPE_ROUTE_MAP_COMPILED, sizeof(struct rmap_community));
 
-	rcom->name = XSTRDUP(MTYPE_ROUTE_MAP_COMPILED, arg);
+	char **splits;
+	char *name = NULL;
+	int num;
+
+	frrstr_split(arg, " ", &splits, &num);
+	if (num > 1)
+		name = XSTRDUP(MTYPE_TMP, splits[0]);
```

Comment:
Is there a particular reason for the double strdup? Why not just copy `splits[0]` directly to `rcom->name`?